src/main/java/io/neonbee/data/internal/metrics/ConfiguredDataVerticleMetrics.java

```diff
@@ -0,0 +1,160 @@
+    public static final String METER_REGISTRY_NAME = "meterRegistryName";
+
+    /**
+     * Key for reporting the request count.
+     */
+    public static final String NUMBER_OF_REQUESTS = "reportNumberOfRequests";
+
+    /**
+     * Key for reporting the currently active requests.
+     */
+    public static final String ACTIVE_REQUESTS = "reportActiveRequests";
+
+    /**
+     * Key for reporting the status counter.
+     */
+    public static final String STATUS_COUNTER = "reportStatusCounter";
+
+    /**
+     * Key for reporting timing values.
+     */
+    public static final String TIMING = "reportTiming";
+
+    @VisibleForTesting
+    static final DummyDataVerticleMetrics DUMMY_IMPL = new DummyDataVerticleMetrics();
+
+    private static final LoggingFacade LOGGER = LoggingFacade.create();
+
+    @VisibleForTesting
+    final DataVerticleMetrics reportNumberOfRequests;
+
+    @VisibleForTesting
+    final DataVerticleMetrics reportActiveRequestsGauge;
+
+    @VisibleForTesting
+    final DataVerticleMetrics reportStatusCounter;
+
+    @VisibleForTesting
+    final DataVerticleMetrics reportTimingMetric;
+
+    ConfiguredDataVerticleMetrics(DataVerticleMetrics reportNumberOfRequests,
+            DataVerticleMetrics reportActiveRequestsGauge, DataVerticleMetrics reportStatusCounter,
+            DataVerticleMetrics reportTimingMetric) {
+        this.reportNumberOfRequests = reportNumberOfRequests;
+        this.reportActiveRequestsGauge = reportActiveRequestsGauge;
+        this.reportStatusCounter = reportStatusCounter;
+        this.reportTimingMetric = reportTimingMetric;
+    }
+
+    /**
+     * Configure the {@link DataVerticleMetrics} to report.
+     *
+     * To enable all metrics, you must specify the enabled key with the value ture. Otherwise, no metrics will be
+     * reported. You can select the registry to use by specifying the meterRegistryName key with the name of the
+     * registry associated with the registry in the micrometer options.
+     *
+     * If you specify any of the configuration values "reportNumberOfRequests", "reportActiveRequests",
+     * "reportStatusCounter", "reportTiming", only the values configured as true will be reported. If you do not specify
+     * any of these values, all metrics are reported.
+     *
+     * Full example:
+     *
+     * <pre>
+     * {@code
+     * {
+     *     "enabled" : true,
+     *     "meterRegistryName" : "default",
+     *     "reportNumberOfRequests" : true,
+     *     "reportActiveRequests" : true
+     *     "reportStatusCounter" : true,
+     *     "reportTiming" : true
+     * }
+     * }
+     * </pre>
+     *
+     * @param metricsConfig {@link JsonObject} containing the metrics configuration.
+     * @return configured {@link DataVerticleMetrics}
+     */
+    public static DataVerticleMetrics configureMetricsReporting(JsonObject metricsConfig) {
+        if (metricsConfig == null || !Boolean.TRUE.equals(metricsConfig.getBoolean(ENABLED))) {
```

Comment:
```suggestion
        if (metricsConfig == null || !metricsConfig.getBoolean(ENABLED, false)) {
```

Reply:
No, because you are already checking if `metricsConfig` is null. The latter part is not evaluated.

Reply:
No, if the JSON object metricsConfig would look like this:
`{
  "enabled": null
}`
then we get a NullPointerException with your suggested code change.

I even wrote a JuntiTest for the case: io.neonbee.data.internal.metrics.ConfiguredDataVerticleMetricsTest#disabledNull

Reply:
ah, you are right :) consider my comment as invalid